src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,159 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { KinveyError } from '../errors';
+import { repositoryProvider } from './repositories';
+import { buildCollectionUrl } from './repositories/utils';
+import {
+  stripTagFromCollectionName,
+  generateEntityId,
+  queryCacheCollectionName,
+  xKivneyRequestStartHeader
+} from './utils';
+
+function getDeltaSetQuery(collectionNameWithTag, query) {
+  const serializedQuery = query ? query.toString() : '';
+
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      const queryCacheQuery = new Query()
+        .equalTo('collectionName', collectionNameWithTag)
+        .and()
+        .equalTo('query', serializedQuery);
+      return offlineRepo.read(queryCacheCollectionName, queryCacheQuery)
```

Comment:
I now remember we went a different route with the `kinvey_sync` collection - it is being scoped like any other collection. Here with the "query cache" collection, we are only scoping the collection name of its "rows". Should we may be unify our approach?

Reply:
I don't understand what you mean. Can you explain more?

Reply:
I mean that when you have a data store with no tag, it uses a collection `kinvey_sync` for the sync queue. If you use a data store with a tag, for the same collection, it uses a collection called `kinvey_sync.<tag-name>` for the sync queue.

The same two cases, regarding the `_QueryCache` collection, would use the same collection. Just the values for the `collection` "column" would be different. The same could have been applied to the `kinvey_sync` collection, but we went a different route. I'm saying we should decide which is better, and a unified behaviour might be a consideration in this decision.